internal_coroutines_test.go

```diff
@@ -1034,3 +1034,53 @@ func TestDecodeFutureChain(t *testing.T) {
 	}
 	require.EqualValues(t, expected, history)
 }
+
+func TestSelectFuture_WithBatchSets(t *testing.T) {
+	var history []string
+	d := newDispatcher(background, func(ctx Context) {
+		future1, settable1 := NewFuture(ctx)
+		future2, settable2 := NewFuture(ctx)
+		future3, settable3 := NewFuture(ctx)
+
+		s := NewSelector(ctx)
```

Comment:
i think it make sense if we look select as reading multiple times to unblock it 3 times. will do a separate CR tomorrow.
@yiminc - Which is the reason we have our own select to have deterministic order.